.clang-format

```diff
@@ -0,0 +1,20 @@
+BasedOnStyle: Google
+BreakBeforeBraces: Linux
+IndentWidth: 4
+PPIndentWidth: 1
+IndentPPDirectives: AfterHash
+IndentCaseLabels: false
+IncludeBlocks: Preserve
+SortIncludes: true
```

Comment:
Not sure how this one works exactly, but is it too strict? Sometimes `#include "config.h"` needs to be first and we usually put system headers first.

Reply:
Includes should be divide in sections. if config has to be first it should be put into own section. This is also described well in our contributing guide https://github.com/mpv-player/mpv/blob/master/DOCS/contribute.md#header-include-statement-order

Reply:
Oh so it works fine if everything is separated by sections? I wasn't sure.

Reply:
Yes, exactly. This is what `IncludeBlocks: Preserve` option does to preserve blocks that are in the code. clang-format is capable of regrouping includes on its own  too, but this would require probably defining custom regexes for our codebase https://clang.llvm.org/docs/ClangFormatStyleOptions.html#includecategories I didn't bother with it. We are not going to format everything anyway. This check will run only on new added code and I wouldn't consider it hard error and instead suggestion how code can be formatted. Mostly to fix indention errors and alike...